### DISCUSSION TITLE
automatic font and button sizing?

### DISCUSSION THREAD

**Discussion by acroyear** [beercss/beercss#257] [Ideas]
My app (or at least, the framework of components for it) has an unusual set of requirements - it should work ok on desktop, mobile, but also TV sets (Amazon Fire Stick, potentially Android TV, and i'm thinking of making a chromecast receiver).

as such, I have varying font requirements based on the target. On a phone, i'm ok with things staying 's' small. On desktop, 'm' is fine unless the user's shrunk the window down to "small" breakpoint at which point i'd want small.

On the TV I need everything L or even 'Extra' if available.  I can't just rely on the screensize (1920) because again the experience can be different on desktop (monitor is only 2 feet away) from a TV (60" tv 10 feet away), so on the desktop situation i'd prefer the 'm' size. :)

so right now to get that, i need to look at the responsive size and set the s/m/l (and extra) sizes based on your breakpoints.

I think a better solution would be to publish the font-size --css-variables (the way the colors are published) so that there's some documentation on how to override them to specialize on these oddball situations where normal responsive rules just aren't enough.

**Comment by laurentpayot** [comment]
I think you can use a [media query](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_media_queries/Using_media_queries) to change all the Beercss element sizes. For instance for a large screen:

```css
@media screen and (min-width: 1400px) {
  :root {
    /* default size is 16px */
    --size: 24px;
  }
}


```

**Reply by acroyear** [reply]
bookmarking.  been working on other things of late so i hadn't gotten back to trying to get this working on tv sets for a while.

**Reply by acroyear** [reply]
and i have an additional fun for also scaling things well for 4K tv sets. i'm finding HUGE gaps on the screen in my players when i'm on my 48" tv :)
